include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -4391,14 +4397,55 @@ namespace internal
 
     template <int fe_degree, int = 0>
     static bool
-    run(const unsigned int,
-        const AlignedVector<Number> &,
-        const Number *,
-        Number *,
+    run(const unsigned int                  n_desired_components,
+        const FEEvaluationBaseData<dim,
+                                   typename Number::value_type,
+                                   false,
+                                   Number> &fe_eval,
+        const Number *                      in_array,
+        Number *                            out_array,
         typename std::enable_if<fe_degree == -1>::type * = nullptr)
     {
       static_assert(fe_degree == -1, "Only usable for degree -1");
-      Assert(false, ExcNotImplemented());
+
+      const AlignedVector<Number> &inverse_shape =
+        fe_eval.get_shape_info().data.front().inverse_shape_values;
+
+      const unsigned int dofs_per_component =
+        fe_eval.get_shape_info().dofs_per_component_on_cell;
+      const unsigned int n_q_points = fe_eval.get_shape_info().n_q_points;
+
+      internal::
+        EvaluatorTensorProduct<internal::evaluate_general, dim, 0, 0, Number>
+          evaluator(AlignedVector<Number>(),
+                    AlignedVector<Number>(),
+                    inverse_shape,
+                    fe_eval.get_shape_info().data.front().n_q_points_1d,
+                    fe_eval.get_shape_info().data.front().fe_degree + 1);
+
+      AlignedVector<Number> temp_1(std::max(n_q_points, dofs_per_component));
+      AlignedVector<Number> temp_2(std::max(n_q_points, dofs_per_component));
```

Comment:
This does not seem right. I think you want to use https://github.com/dealii/dealii/blob/2fd0c22c0c871a60de9c62e81e87a754d3dc2fbe/include/deal.II/matrix_free/fe_evaluation.h#L244-L251

Reply:
Done!